### PR TITLE
Fix issues reported by cppcheck

### PR DESCRIPTION
**Pull request by masterleinad**
These are the reasonable changes I found after running `cppcheck`. In reference to #6156.

### CODE REVIEW

**Comment by tamiko** [comment]
@masterleinad Next time we should synchronize a bit :-D, I will not make a pull request for the rest I have locally it seems. Also have a quick look at one minor change, https://github.com/dealii/dealii/pull/6345

**Comment by drwells** [comment]
Its probably more trouble than its worth, but we could probably replace this bessel function with one from boost and ignore the configuration issue completely.

**Comment by bangerth** [comment]
It's definitely not worth doing before the release. But if you wanted to open a github issue, then we can address that at some unspecified point in the future.

**Comment by kronbichler** [comment]
Let us merge this now - I do not think that waiting for the issue should stop this here.

File: source/grid/grid_in.cc [hunk 1]
```diff
@@ -1640,7 +1640,6 @@ void GridIn<2>::read_netcdf (const std::string &filename)
 #else
   const unsigned int dim=2;
   const unsigned int spacedim=2;
-  const bool output=false;
```
Comment (kronbichler) [inline source/grid/grid_in.cc]: @masterleinad There is a number of uses of `output` that you apparently forgot to remove, see the build errors here:
https://cdash.kyomu.43-1.org/viewBuildError.php?buildid=1130

Could you please have a look?

**Pull request merged**